MovieCave - MVVM/MovieCave/MovieCaveTests/DetailsScreenViewModelTests.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  DetailsScreenViewModelTests.swift
+//  MovieCaveTests
+//
+//  Created by Admin on 1.12.23.
+//
+
+import XCTest
+@testable import MovieCave
+
+final class DetailsScreenViewModelTests: XCTestCase {
+    
+    var mockMovieDBService: MovieDBManagerMock!
+    var movieDetailsCoordinator: MovieDetailsViewCoordinator!
+    var tvSeriesDetailsCoordinator: TVSeriesDetailsCoordinator!
+    var viewModel: MoviesDetailsViewModel!
+    let navController = UINavigationController()
+    
+    override func setUp() {
+        super.setUp()
+        mockMovieDBService = MovieDBManagerMock(succesCase: .happy, moviesList: .allMovies)
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        mockMovieDBService = nil
+        movieDetailsCoordinator = nil
+        tvSeriesDetailsCoordinator = nil
+        viewModel = nil
+    }
+    
+    func test_DetailsView_withTVSeries_HappyCase() {
+        // Given
+        let id = 10
+        movieDetailsCoordinator = MovieDetailsViewCoordinator(navController: navController, id: id)
+        viewModel = MoviesDetailsViewModel(mediaType: .tvSeries, mediaID: id, coordinator: movieDetailsCoordinator, apiService: mockMovieDBService)
+        
+        // When
+        
+        
+        // Then
+        XCTAssertEqual(viewModel.mediaDetails.value, mockMovieDBService.mediaDetails)
+        XCTAssertEqual(viewModel.mediaVideos.value?.last, mockMovieDBService.mediaVideos)
+        XCTAssertEqual(viewModel.mediaCast.value?.last, mockMovieDBService.mediaCast)
+        XCTAssertNil(mockMovieDBService.apiCallError)
+        XCTAssertNotNil(mockMovieDBService.apiCallTVSeriesResult?.results)
+        XCTAssertNotNil(viewModel.mediaCast.value)
+        XCTAssertNotNil(viewModel.mediaVideos.value)
+        XCTAssertNotNil(viewModel.mediaDetails.value)
+    }
+    
+    func test_DetailsView_withTVSeries_SadCase() {
+        // Given
+        let id = 10
+        movieDetailsCoordinator = MovieDetailsViewCoordinator(navController: navController, id: id)
+        mockMovieDBService.succesCase = .sad
+        viewModel = MoviesDetailsViewModel(mediaType: .tvSeries, mediaID: id, coordinator: movieDetailsCoordinator, apiService: mockMovieDBService)
+        
+        // When
+        
```

Comment:
you could add the initializing of the view model to the // When. Do the same for the rest of the tests if it makes sense